src/stores/blockstore.js

```diff
@@ -37,12 +41,15 @@ exports.setUp = (basePath, BlobStore, locks) => {
     const key = multihashToPath(blockBlob.key)
 
     lock(key, (release) => {
-      pull(
-        pull.values([
-          blockBlob.data
-        ]),
-        store.write(key, release(released))
-      )
+      // write to each store in parallel
+      each(stores, (store, cb) => {
+        pull(
+          pull.values([
+            blockBlob.data
+          ]),
+          store.write(key, cb)
+        )
+      }, release(released))
```

Comment:
if this is true then it seems like the locking should be implemented at the blobStore level